src/components/Fuel/FuelForm.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  FuelSourceUnit,
+  FuelSourceType,
+  UnregisteredFuelRequest,
+} from '../../schema/fuel.schema';
+
+interface FuelFormProps {
+  handleSubmit: (fuelData: UnregisteredFuelRequest) => void;
+}
+
+export default function FuelForm({ handleSubmit }: FuelFormProps) {
+  /* Initializing state */
+  const [fuelData, setfuelData] = useState({
+    fuel_source_type: 'ng' as FuelSourceType,
+    fuel_source_unit: 'btu' as FuelSourceUnit,
+    fuel_source_value: 0 as number,
+  });
+
+  function submitForm(e: React.FormEvent) {
+    e.preventDefault();
+    console.log(fuelData);
+    handleSubmit(fuelData);
+  }
+
+  /* Function that updates the state */
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    /* Destructuring the event.target object */
+    const { name, value, valueAsNumber } = event?.target;
+    setfuelData((prevFuelData) => {
+      return {
+        ...prevFuelData,
+        [name]: name == 'fuel_source_value' ? valueAsNumber : value,
+      };
+    });
+  }
+
+  /* Renders the units conditionally */
+  const unitOptions =
+    fuelData.fuel_source_type == 'ng'
+      ? ['thousand_cubic_feet', 'btu']
+      : ['gallon', 'btu'];
+
+  function resetForm() {
+    setfuelData({
+      fuel_source_type: 'ng',
+      fuel_source_unit: 'btu',
+      fuel_source_value: 0,
+    });
+  }
+
+  return (
+    <div>
+      <Label htmlFor="fuel_source_type">Which fuel do you use?</Label>
+      <Select
+        id="fuel_source_type"
+        value={fuelData.fuel_source_type}
+        onChange={handleChange}
+        name="fuel_source_type"
+      >
+        <option value="">--- Choose</option>
+        <option value="ng">Natural Gas</option>
+        <option value="dfo">Home Heating and Diesel Fuel</option>
+        <option value="pg">Propane Gas</option>
+        <option value="ker">Kerosene</option>
+      </Select>
+
+      <br />
+
+      <Label htmlFor="unit">Which measuring unit do you use?</Label>
+      <Select
+        id="fuel_source_unit"
+        value={fuelData.fuel_source_unit}
+        onChange={handleChange}
+        name="fuel_source_unit"
+      >
+        <option value="">-- Choose</option>
+        <option value={unitOptions[0]}>{unitOptions[0]}</option>
+        <option value={unitOptions[1]}>{unitOptions[1]}</option>
+      </Select>
+
+      <br />
+
+      <Label htmlFor="value">
+        What is the quantity of fuel you consumed? (based on the unit
+        you&apos;ve chosen above)
+      </Label>
+      <TextInput
+        id="fuel_source_value"
+        type="number"
+        value={fuelData.fuel_source_value}
+        onChange={handleChange}
+        name="fuel_source_value"
+      />
+
+      <div>
+        <Button color="info" onClick={resetForm}>
+          Reset
+        </Button>
+
+        <Button
+          type="submit"
+          disabled={
+            !fuelData.fuel_source_type ||
+            !fuelData.fuel_source_unit ||
+            fuelData.fuel_source_value < 1
```

Comment:
You don't need the ternary here it can just be `fuelData.fuel_source_value < 1` since it evaluates to a boolean

Reply:
Amanda's fine on this. it's not a ternary, but a part of a boolean expression. It's valid.

Reply:
Sorry I wasn't clear, I'm talking about this ternary `fuelData.fuel_source_value < 1 ? true : false` this is redundant since `fuelData.fuel_source_value < 1` will evaluate to true or false without explicitly stating it with a ternary

Reply:
Got that!